### PR TITLE
Prepare for 1.4.0 release

### PR DESCRIPTION
**Pull request by tgraf**
* PR: 6978 -- Added copy buttons for code blocks in docs (@geakstr) -- https://github.com/cilium/cilium/pull/6978
 * PR: 6988 -- cilium, sockmap: unbreak --sockmap-enable mode (@borkmann) -- https://github.com/cilium/cilium/pull/6988
 * PR: 6990 -- examples/kubernetes: add containerd kubernetes files (@aanm) -- https://github.com/cilium/cilium/pull/6990
 * PR: 6980 -- endpoint: make "Regenerating endpoint" log Debug level (@ianvernon) -- https://github.com/cilium/cilium/pull/6980
```
$ for pr in 6988 6978 6990 6980; do contrib/backporting/set-labels.py $pr done 1.4; done
```

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6994)

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-docs-please

**Comment by tgraf** [comment]
test-missed-k8s

File: examples/kubernetes/1.9/cilium.yaml [hunk 1]
```diff
@@ -554,7 +554,7 @@ spec:
               key: AWS_DEFAULT_REGION
               name: cilium-aws
               optional: true
-        image: docker.io/cilium/operator:v1.4.0-rc8
+        image: docker.io/cilium/operator:1.4.0
```
Comment (aanm) [inline examples/kubernetes/1.9/cilium.yaml]: `v1.4.0` not `1.4.0`

**Review by aanm — CHANGES_REQUESTED** [review]

**Comment by aanm** [comment]
@tgraf is it too late to backport https://github.com/cilium/cilium/pull/6990?

**Comment by tgraf** [comment]
> @tgraf is it too late to backport #6990?

No, we can add it. I'll rebase after the tests have passed and also fix the version.

**Comment by tgraf** [comment]
test-docs-please

**Comment by tgraf** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-missed-k8s

**Comment by ianvernon** [comment]
Build failure: https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-K8s/1635/execution/node/56/log/?consoleFull

```
19:54:56 K8sKafkaPolicyTest Kafka Policy Tests 
19:54:56   KafkaPolicies
19:54:56   /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-K8s/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:408
19:55:03 STEP: Installing DNS Deployment
19:55:04 STEP: Deploying etcd-operator
19:55:08 STEP: Installing Cilium
19:55:14 STEP: Waiting for all etcd-operator pods are ready
19:55:19 STEP: Performing Cilium preflight check
19:55:19 STEP: Checking that cilium status is healthy
19:55:29 STEP: Checking that no controllers in Cilium have failed
19:55:37 STEP: Checking that cilium-health status is healthy
19:56:07 STEP: Checking that Kubernetes services are correctly plumbed by Cilium
19:56:07 STEP: Checking that the default/kubernetes service is correctly plumbed by Cilium
19:56:23 STEP: Checking that the kube-system/kube-dns service is correctly plumbed by Cilium
20:01:49 ===================== TEST FAILED =====================
20:01:49 cmd: kubectl get pods -o wide --all-namespaces
20:01:49 Exitcode: 0 
20:01:49 Stdout:
20:01:49  	 NAMESPACE     NAME                                    READY     STATUS              RESTARTS   AGE       IP              NODE
20:01:49 	 default       empire-backup-7d9486fb7c-p685j          0/1       ContainerCreating   0          5m        <none>          k8s1
20:01:49 	 default       empire-hq-6f98fd55c7-g5fn2              1/1       Running             0          5m        10.10.0.34      k8s1
20:01:49 	 default       empire-outpost-8888-68685c979f-l2fpk    1/1       Running             0          5m        10.10.1.152     k8s2
20:01:49 	 default       empire-outpost-9999-5b6b7bccc9-vtfpq    1/1       Running             0          5m        10.10.1.161     k8s2
20:01:49 	 default       kafka-broker-5d5df48574-zc5gx           1/1       Running             0          5m        10.10.1.128     k8s2
20:01:49 	 kube-system   cilium-4twgj                            1/1       Running             2          13m       192.168.36.11   k8s1
20:01:49 	 kube-system   cilium-etcd-5mz6gqb7pz                  1/1       Running             0          9m        10.10.1.133     k8s2
20:01:49 	 kube-system   cilium-etcd-operator-5cd8fb8b78-z4v2l   1/1       Running             0          14m       192.168.36.12   k8s2
20:01:49 	 kube-system   cilium-etcd-rjd2fs9bsz                  1/1       Running             0          11m       10.10.1.3       k8s2
20:01:49 	 kube-system   cilium-etcd-rmv6fh87vt                  1/1       Running             0          10m       10.10.0.60      k8s1
20:01:49 	 kube-system   cilium-kbml7                            1/1       Running             3          13m       192.168.36.12   k8s2
20:01:49 	 kube-system   etcd-k8s1                               1/1       Running             0          30m       192.168.36.11   k8s1
20:01:49 	 kube-system   etcd-operator-58cf6d756d-djl6v          1/1       Running             0          13m       10.10.1.208     k8s2
20:01:49 	 kube-system   kube-apiserver-k8s1                     1/1       Running             0          31m       192.168.36.11   k8s1
20:01:49 	 kube-system   kube-controller-manager-k8s1            1/1       Running             0          31m       192.168.36.11   k8s1
20:01:49 	 kube-system   kube-dns-f4d788bb7-58x7d                3/3       Running             0          31m       10.10.1.187     k8s2
20:01:49 	 kube-system   kube-proxy-7dxrj                        1/1       Running             0          14m       192.168.36.12   k8s2
20:01:49 	 kube-system   kube-proxy-zqzq2                        1/1       Running             0          31m       192.168.36.11   k8s1
20:01:49 	 kube-system   kube-scheduler-k8s1                     1/1       Running             0          31m       192.168.36.11   k8s1
20:01:49 	 
20:01:49 Stderr:
20:01:49  	 
20:01:49 
20:01:51 cmd: kubectl exec -n kube-system cilium-4twgj -- cilium service list
20:01:51 Exitcode: 0 
20:01:51 Stdout:
20:01:51  	 ID   Frontend              Backend                   
20:01:51 	 1    10.96.0.10:53         1 => 10.10.1.187:53       
20:01:51 	 2    10.96.0.1:443         1 => 192.168.36.11:6443   
20:01:51 	 3    10.110.116.214:2379   1 => 10.10.0.60:2379      
20:01:51 	                            2 => 10.10.1.133:2379     
20:01:51 	                            3 => 10.10.1.3:2379       
20:01:51 	 
20:01:51 Stderr:
20:01:51  	 
20:01:51 
20:01:56 cmd: kubectl exec -n kube-system cilium-4twgj -- cilium endpoint list
20:01:56 Exitcode: 0 
20:01:56 Stdout:
20:01:56  	 ENDPOINT   POLICY (ingress)   POLICY (egress)   IDENTITY   LABELS (source:key[=value])                       IPv6                 IPv4          STATUS   
20:01:56 	            ENFORCEMENT        ENFORCEMENT                                                                                                       
20:01:56 	 878        Disabled           Disabled          33279      k8s:app=empire-backup                             f00d::a0a:0:0:9261   10.10.0.19    ready   
20:01:56 	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                     
20:01:56 	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                              
20:01:56 	                                                            k8s:io.kubernetes.pod.namespace=default                                                      
20:01:56 	                                                            k8s:zgroup=kafkaTestApp                                                                      
20:01:56 	 1491       Disabled           Disabled          4          reserved:health                                   f00d::a0a:0:0:56b4   10.10.0.247   ready   
20:01:56 	 1889       Disabled           Disabled          3614       k8s:app=empire-hq                                 f00d::a0a:0:0:9496   10.10.0.34    ready   
20:01:56 	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                     
20:01:56 	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                              
20:01:56 	                                                            k8s:io.kubernetes.pod.namespace=default                                                      
20:01:56 	                                                            k8s:zgroup=kafkaTestApp                                                                      
20:01:56 	 1905       Disabled           Disabled          101        k8s:app=etcd                                      f00d::a0a:0:0:1c03   10.10.0.60    ready   
20:01:56 	                                                            k8s:etcd_cluster=cilium-etcd                                                                 
20:01:56 	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                     
20:01:56 	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                              
20:01:56 	                                                            k8s:io.cilium/app=etcd-operator                                                              
20:01:56 	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                  
20:01:56 	 
20:01:56 Stderr:
20:01:56  	 
20:01:56 
20:01:58 cmd: kubectl exec -n kube-system cilium-kbml7 -- cilium service list
20:01:58 Exitcode: 0 
20:01:58 Stdout:
20:01:58  	 ID   Frontend              Backend                   
20:01:58 	 1    10.96.0.1:443         1 => 192.168.36.11:6443   
20:01:58 	 2    10.96.0.10:53         1 => 10.10.1.187:53       
20:01:58 	 3    10.110.116.214:2379   1 => 10.10.1.133:2379     
20:01:58 	                            2 => 10.10.1.3:2379       
20:01:58 	                            3 => 10.10.0.60:2379      
20:01:58 	 
20:01:58 Stderr:
20:01:58  	 
20:01:58 
20:02:02 cmd: kubectl exec -n kube-system cilium-kbml7 -- cilium endpoint list
20:02:02 Exitcode: 0 
20:02:02 Stdout:
20:02:02  	 ENDPOINT   POLICY (ingress)   POLICY (egress)   IDENTITY   LABELS (source:key[=value])                              IPv6                   IPv4          STATUS   
20:02:02 	            ENFORCEMENT        ENFORCEMENT                                                                                                                
20:02:02 	 647        Disabled           Disabled          102        k8s:io.cilium.k8s.policy.cluster=default                 f00d::a0a:100:0:12a1   10.10.1.187   ready   
20:02:02 	                                                            k8s:io.cilium.k8s.policy.serviceaccount=kube-dns                                                      
20:02:02 	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                           
20:02:02 	                                                            k8s:k8s-app=kube-dns                                                                                  
20:02:02 	 839        Disabled           Disabled          101        k8s:app=etcd                                             f00d::a0a:100:0:b7a6   10.10.1.3     ready   
20:02:02 	                                                            k8s:etcd_cluster=cilium-etcd                                                                          
20:02:02 	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                              
20:02:02 	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                       
20:02:02 	                                                            k8s:io.cilium/app=etcd-operator                                                                       
20:02:02 	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                           
20:02:02 	 985        Disabled           Disabled          51662      k8s:app=empire-outpost                                   f00d::a0a:100:0:f9b2   10.10.1.152   ready   
20:02:02 	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                              
20:02:02 	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                       
20:02:02 	                                                            k8s:io.kubernetes.pod.namespace=default                                                               
20:02:02 	                                                            k8s:outpostid=8888                                                                                    
20:02:02 	                                                            k8s:zgroup=kafkaTestApp                                                                               
20:02:02 	 1147       Disabled           Disabled          4          reserved:health                                          f00d::a0a:100:0:9498   10.10.1.188   ready   
20:02:02 	 1205       Disabled           Disabled          101        k8s:app=etcd                                             f00d::a0a:100:0:457a   10.10.1.133   ready   
20:02:02 	                                                            k8s:etcd_cluster=cilium-etcd                                                                          
20:02:02 	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                              
20:02:02 	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                       
20:02:02 	                                                            k8s:io.cilium/app=etcd-operator                                                                       
20:02:02 	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                           
20:02:02 	 3504       Disabled           Disabled          51631      k8s:app=kafka                                            f00d::a0a:100:0:523e   10.10.1.128   ready   
20:02:02 	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                              
20:02:02 	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                       
20:02:02 	                                                            k8s:io.kubernetes.pod.namespace=default                                                               
20:02:02 	                                                            k8s:zgroup=kafkaTestApp                                                                               
20:02:02 	 3549       Disabled           Disabled          100        k8s:io.cilium.k8s.policy.cluster=default                 f00d::a0a:100:0:76f7   10.10.1.208   ready   
20:02:02 	                                                            k8s:io.cilium.k8s.policy.serviceaccount=cilium-etcd-sa                                                
20:02:02 	                                                            k8s:io.cilium/app=etcd-operator                                                                       
20:02:02 	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                           
20:02:02 	 3733       Disabled           Disabled          41074      k8s:app=empire-outpost                                   f00d::a0a:100:0:7e97   10.10.1.161   ready   
20:02:02 	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                              
20:02:02 	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                       
20:02:02 	                                                            k8s:io.kubernetes.pod.namespace=default                                                               
20:02:02 	                                                            k8s:outpostid=9999                                                                                    
20:02:02 	                                                            k8s:zgroup=kafkaTestApp                                                                               
20:02:02 	 
20:02:02 Stderr:
20:02:02  	 
20:02:02 
20:05:55 ===================== Existing AfterFailed =====================
20:05:55 <Checks>
20:05:55 ⚠️  Found a 'JoinEP: ' in logs
20:05:55 Number of 'context deadline exceeded' in logs: 0
20:05:55 ⚠️  Number of 'level=error' in logs: 15
20:05:55 ⚠️  Number of 'level=warning' in logs: 31
20:05:55 Number of 'Cilium API handler panicked' in logs: 0
20:05:55 ⚠️  Number of 'Goroutine took lock for more than' in logs: 150
20:05:55 Cilium pods: [cilium-4twgj cilium-kbml7]
20:05:55 Netpols loaded: 
20:05:55 CiliumNetworkPolicies loaded: 
20:05:55 Endpoint Policy Enforcement:
20:05:55 Pod                                    Ingress   Egress
20:05:55 empire-outpost-9999-5b6b7bccc9-vtfpq   false     false
20:05:55 kafka-broker-5d5df48574-zc5gx          false     false
20:05:55 cilium-etcd-rjd2fs9bsz                 false     false
20:05:55 etcd-operator-58cf6d756d-djl6v         false     false
20:05:55 kube-dns-f4d788bb7-58x7d               false     false
20:05:55 empire-hq-6f98fd55c7-g5fn2             false     false
20:05:55 empire-outpost-8888-68685c979f-l2fpk   false     false
20:05:55 cilium-etcd-5mz6gqb7pz                 false     false
20:05:55 cilium-etcd-rmv6fh87vt                 false     false
20:05:55 Cilium agent 'cilium-4twgj': Status: Ok  Health: Ok Nodes "k8s1 k8s2" ContinerRuntime: Ok Kubernetes: Ok KVstore: Ok Controllers: Total 33 Failed 0
20:05:55 Cilium agent 'cilium-kbml7': Status: Ok  Health: Ok Nodes "k8s1 k8s2" ContinerRuntime: Ok Kubernetes: Ok KVstore: Ok Controllers: Total 54 Failed 0
20:05:55 
20:05:55 </Checks>
20:05:55 
20:05:59 [[ATTACHMENT|57d11463_K8sKafkaPolicyTest_Kafka_Policy_Tests_KafkaPolicies.zip]]
20:05:59 
20:05:59 • Failure in Spec Setup (BeforeEach) [663.476 seconds]
20:05:59 K8sKafkaPolicyTest
20:05:59 /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-K8s/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:376
20:05:59   Kafka Policy Tests
20:05:59   /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-K8s/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:376
20:05:59     KafkaPolicies [BeforeEach]
20:05:59     /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-K8s/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:408
20:05:59 
20:05:59     Kafka Pods are not ready after timeout
20:05:59     Expected
20:05:59         <*errors.errorString | 0xc420889a30>: {
20:05:59             s: "Timeout reached: timed out waiting for pods with filter -l zgroup=kafkaTestApp to be ready",
20:05:59         }
20:05:59     to be nil
20:05:59 
20:05:59     /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-K8s/src/github.com/cilium/cilium/test/k8sT/KafkaPolicies.go:129
20:05:59 ------------------------------
```

Both Cilium pods got into a bad state because they couldn't connect to the kvstore:
```
2019-02-08T19:48:25.97690098Z level=fatal msg="Unable to connect to kvstore" error="open /var/lib/etcd-secrets/etcd-client.crt: no such file or directory" subsys=kvstore
```

Logs contain `JoinEP` per the above output:

```
pod-kube-system-cilium-4twgj-cilium-agent.log:2019-02-08T20:00:30.511379637Z level=warning msg="JoinEP: Failed to load program" containerID=a58768238a datapathPolicyRevision=0 desiredPolicyRevision=6 endpointID=2185 error="Failed to replace Qdisc for lxcb0ff8a5f5567: Link not found" file-path=2185_next/bpf_lxc.o identity=33279 ipv4=10.10.0.85 ipv6="f00d::a0a:0:0:e4" k8sPodName=default/empire-backup-7d9486fb7c-p685j subsys=datapath-loader veth=lxcb0ff8a5f5567
pod-kube-system-cilium-4twgj-cilium-agent.log:2019-02-08T20:01:28.737926465Z level=warning msg="JoinEP: Failed to load program" containerID=d6c238cf05 datapathPolicyRevision=0 desiredPolicyRevision=6 endpointID=4052 error="Failed to load tc filter: exit status 2" file-path=4052_next/bpf_lxc.o identity=33279 ipv4=10.10.0.42 ipv6="f00d::a0a:0:0:54b8" k8sPodName=default/empire-backup-7d9486fb7c-p685j subsys=datapath-loader veth=lxca4ee78e7c684
pod-kube-system-cilium-kbml7-cilium-agent.log:2019-02-08T19:58:07.758899626Z level=warning msg="JoinEP: Failed to load program" containerID=b463cd119d datapathPolicyRevision=0 desiredPolicyRevision=6 endpointID=880 error="Failed to load tc filter: exit status 2" file-path=880_next/bpf_lxc.o identity=51662 ipv4=10.10.1.150 ipv6="f00d::a0a:100:0:8268" k8sPodName=default/empire-outpost-8888-68685c979f-l2fpk subsys=datapath-loader veth=lxcec0e62655650
```

Looks related to 
https://github.com/cilium/cilium/issues/6059 .

**Comment by ianvernon** [comment]
test-missed-k8s

**Pull request merged**